dbt_airflow_manifest_parser/builder.py

```diff
@@ -27,21 +33,26 @@ def _make_dbt_run_task(self, model_name: str) -> BaseOperator:
     def _is_model_run_task(node_name: str):
         return node_name.split(".")[0] == "model"
 
-    def _create_tasks_for_each_model(self, manifest: dict) -> dict:
+    def _create_tasks_for_each_model(self, manifest: dict, use_task_group: bool) -> dict:
         tasks = {}
         for node_name in manifest["nodes"].keys():
             if self._is_model_run_task(node_name):
                 logging.info("Creating tasks for: " + node_name)
                 model_name = node_name.split(".")[-1]
-                run_task = self._make_dbt_run_task(model_name)
-                test_task = self._make_dbt_test_task(model_name)
-                # noinspection PyStatementEffect
-                run_task >> test_task
-                tasks[node_name] = ModelExecutionTask(run_task, test_task)
+
+                is_first_version = airflow.__version__.startswith("1.")
+                task_group = None if (is_first_version or not use_task_group) else TaskGroup(group_id=model_name)
+                task_group_ctx = task_group or contextlib.nullcontext()
```

Comment:
could you extract these 3 lines into separated method called eg: create_task_group

Reply:
done